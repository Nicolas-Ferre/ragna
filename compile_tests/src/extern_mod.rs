fn main() {}

#[ragna::gpu]
mod gpu {
    extern "C" {}

    extern "wgsl" {
        static STATIC: u32;
    }
}
