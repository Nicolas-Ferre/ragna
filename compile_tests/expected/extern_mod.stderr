error: only "wgsl" ABI is supported
 --> compile_tests/src/extern_mod.rs:7:12
  |
7 |     extern "C" {}
  |            ^^^

error: unsupported item
  --> compile_tests/src/extern_mod.rs:10:9
   |
10 |         static STATIC: U32;
   |         ^^^^^^

error: function must have a return type
  --> compile_tests/src/extern_mod.rs:12:12
   |
12 |         fn no_return_type();
   |            ^^^^^^^^^^^^^^

error: unsupported parameter
  --> compile_tests/src/extern_mod.rs:14:27
   |
14 |         fn func_with_self(self) -> F32;
   |                           ^^^^

error: `self` parameter is only allowed in associated functions
  --> compile_tests/src/extern_mod.rs:14:27
   |
14 |         fn func_with_self(self) -> F32;
   |                           ^^^^ not semantically valid as function parameter
   |
   = note: associated functions are those in `impl` or `trait` definitions
